### ISSUE TITLE
Signature Version 4 for eu-central-1

### ISSUE THREAD

**Issue by jakobjuelich** [rslifka/elasticity#83]
When trying to start a job flow on eu-central-1 I get the following exception:

```
RestClient::Forbidden (403 Forbidden):
    /home/xyz/snowplow/3-enrich/emr-etl-runner/vendor/bundle/ruby/1.9.1/gems/rest-client-1.7.3/lib/restclient/abstract_response.rb:48:in `return!'
    /home/xyz/snowplow/3-enrich/emr-etl-runner/vendor/bundle/ruby/1.9.1/gems/rest-client-1.7.3/lib/restclient/request.rb:497:in `process_result'
    /home/xyz/snowplow/3-enrich/emr-etl-runner/vendor/bundle/ruby/1.9.1/gems/rest-client-1.7.3/lib/restclient/request.rb:421:in `block in transmit'
    /usr/lib/ruby/1.9.1/net/http.rb:746:in `start'
    /home/xyz/snowplow/3-enrich/emr-etl-runner/vendor/bundle/ruby/1.9.1/gems/rest-client-1.7.3/lib/restclient/request.rb:413:in `transmit'
    /home/xyz/snowplow/3-enrich/emr-etl-runner/vendor/bundle/ruby/1.9.1/gems/rest-client-1.7.3/lib/restclient/request.rb:176:in `execute'
    /home/xyz/snowplow/3-enrich/emr-etl-runner/vendor/bundle/ruby/1.9.1/gems/rest-client-1.7.3/lib/restclient/request.rb:41:in `execute'
    /home/xyz/snowplow/3-enrich/emr-etl-runner/vendor/bundle/ruby/1.9.1/gems/rest-client-1.7.3/lib/restclient.rb:69:in `post'
    /home/xyz/snowplow/3-enrich/emr-etl-runner/vendor/bundle/ruby/1.9.1/gems/elasticity-4.0.5/lib/elasticity/aws_request.rb:35:in `submit'
    /home/xyz/snowplow/3-enrich/emr-etl-runner/vendor/bundle/ruby/1.9.1/gems/elasticity-4.0.5/lib/elasticity/emr.rb:191:in `run_job_flow'
    /home/xyz/snowplow/3-enrich/emr-etl-runner/vendor/bundle/ruby/1.9.1/gems/elasticity-4.0.5/lib/elasticity/job_flow.rb:147:in `run'
    /home/xyz/snowplow/3-enrich/emr-etl-runner/lib/snowplow-emr-etl-runner/emr_job.rb:294:in `run'
    /home/xyz/snowplow/3-enrich/emr-etl-runner/vendor/bundle/ruby/1.9.1/gems/contracts-0.7/lib/contracts/method_reference.rb:46:in `send_to'
    /home/xyz/snowplow/3-enrich/emr-etl-runner/vendor/bundle/ruby/1.9.1/gems/contracts-0.7/lib/contracts.rb:305:in `call_with'
    /home/xyz/snowplow/3-enrich/emr-etl-runner/vendor/bundle/ruby/1.9.1/gems/contracts-0.7/lib/contracts/decorators.rb:159:in `block in common_method_added'
    /home/xyz/snowplow/3-enrich/emr-etl-runner/lib/snowplow-emr-etl-runner/runner.rb:60:in `run'
    /home/xyz/snowplow/3-enrich/emr-etl-runner/vendor/bundle/ruby/1.9.1/gems/contracts-0.7/lib/contracts/method_reference.rb:46:in `send_to'
    /home/xyz/snowplow/3-enrich/emr-etl-runner/vendor/bundle/ruby/1.9.1/gems/contracts-0.7/lib/contracts.rb:305:in `call_with'
    /home/xyz/snowplow/3-enrich/emr-etl-runner/vendor/bundle/ruby/1.9.1/gems/contracts-0.7/lib/contracts/decorators.rb:159:in `block in common_method_added'
    bin/snowplow-emr-etl-runner:39:in `<main>'
```

I digged a little into the code and just inserted a puts res.body into ```rest-client-1.7.3/lib/restclient/request.rb``` to see the actual response. Here it is:

```
<ErrorResponse xmlns="http://elasticmapreduce.amazonaws.com/doc/2009-03-31">
  <Error>
    <Type>Sender</Type>
    <Code>MissingAuthenticationToken</Code>
    <Message>Missing Authentication Token</Message>
  </Error>
  <RequestId>d46bb15a-e9d9-11e4-9030-
```

I believe the problem might be the signature version 4, which is required for eu-central-1. I found a bug that describes the same problem for boto here:
https://github.com/boto/boto/pull/2719

And I checked, in aws_request.rb it seems like SignatureVersion is hard coded to 2.

I'd appreciate very much to if you could take a look ;) Thanks

**Comment by rslifka** [comment]
Absolutely sir, will do ASAP.  Thanks for filing this!

**Comment by rslifka** [comment]
Just a heads up - this is a bit of a beast, will need a bit of time to sort through it.  Will get to it over the next couple of days / this weekend.

**Comment by jakobjuelich** [comment]
Thanks for the quick reaction, I look forward to a fix.

**Comment by rslifka** [comment]
Hi Jakob!  Taking a bit longer than I anticipated, will have something out this week.

**Comment by rslifka** [comment]
Fixed as of 7714af0df5bf2857c2a2cdf7c9672679a5a197ac !

**Comment by jakobjuelich** [comment]
Awesome, thanks!
